src/catapult/catapult.ml

```diff
@@ -80,32 +149,31 @@ let on_process_end t (id, name) =
     {|{"cat": "process", "name": %S, "id": %d, "pid": 0, "ph": "e", "ts": %s}|}
     name id (pp_time time)
 
-let gen_emit_counter t key pvalue value =
+let emit_counter t key values =
   let time = t.get_time () in
-  printf t
-    {|{"name": %S, "pid": 0, "tid": 0, "ph": "C", "ts": %s, "args": {%S: %a}}|}
-    key (pp_time time) "value" pvalue value
-
-let emit_counter t key value =
-  gen_emit_counter t key (Fun.const Int.to_string) value
-
-let emit_counter_float =
-  let float () f = Printf.sprintf "%.2f" f in
-  fun t key value -> gen_emit_counter t key float value
+  let values =
+    Buffer.clear t.buffer;
+    Json.to_buf (Json.Object values) t.buffer;
+    Buffer.contents t.buffer
+  in
+  printf t {|{"name": %S, "pid": 0, "tid": 0, "ph": "C", "ts": %s, "args": %s}|}
```

Comment:
Did you check if an arbitrary json value is allowed for `"args"`? The documentation wasn't very clear on this.